content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,226 @@
+        - `name` — containing a string
+        - `adapter` — containing a string
+        - `host` — also holds a string
+        - `port` — holds an integer
+        - `user` — a string
+        - `secretRef` — containing a map with one key:
+            - `name` — containing a string
+
+## Inferring Default Values
+
+The exact values specified in Schema are the defaults (with one notable exception — arrays — which are [detailed below](#inferring-defaults-for-arrays)). This ensures that all nodes are initialized with values, with reasonable defaults.
+
+Configuration Consumers should specify in their data values file only those values which vary from the default, reducing risk of errors and generally making configuration easier to maintain.
+
+### Inferring Defaults for Scalars
+
+When a scalar value is specified, it is simply the default.  From the example, above:
+
+- `system_domain` is an empty string, by default.
+- `load_balancer.enable` is true by default.
+- `databases[].adapter` is `"postgres"`, by default.
+
+### Inferring Defaults for Mappings
+
+The set of items of a map are its default: any item missing will be automatically inserted with _its_ defaults.
+
+- if `load_balancer` is omitted from supplied Data Values, it is inserted:
+    - `load_balancer.enabled` is `true`
+    - `load_balancer.static_ip` is `""`
+- if `load_balancer` is specified with only `static_ip`, `enabled` is `true`.
+
+When the Configuration Consumer supplies an empty map (i.e. `{}`), all items are defaulted:
+
+For example,
+
+`load_balancer: {}`
+
+is defaulted to:
+
+```yaml
+load_balancer:
+  enable: true
+  static_ip: ""
+```
+
+### Inferring Defaults for Arrays
+
+The default value for all arrays is an empty array. Arrays are the only type for which defaults are **not** inferred.
+
+From the example, above,
+
+- `app_domains`, if not otherwise specified is `[]`
+- `databases`, if not otherwise specified is `[]`
+
+When values for an array _are_ provided by the Configuration Consumer, each item is defaulted based on the type specified in the schema.
+
+Focusing on just `databases`, if the Configuration Consumer supplies this Data Value:
+
+```yaml
+#@data/values
+---
+databases:
+- name: uaa
+- name: capi
+  host: capi-db.svc.cluster.local
+  secretRef:
+    name: capi-db-credentials
+- {}
+```
+
+The resulting Data Values (i.e. that are supplied to templates), would be:
+
+```yaml
+databases:
+- name: uaa
+  adapter: postgresql
+  host: ""
+  port: 5432
+  user: admin
+  secretRef:
+    name: ""
+- name: capi
+  adapter: postgresql
+  host: capi-db.svc.cluster.local
+  port: 5432
+  user: admin
+  secretRef:
+    name: capi-db-credentials
+- name: ""
+  adapter: postgresql
+  host: ""
+  port: 5432
+  user: admin
+  secretRef:
+    name: ""
+```
+
+## Defining Schema Explicitly
+
+Configuration Authors may override inferred typing and defaults through annotations.
+
+### @schema/type
+
+Explicitly sets the type of the annotated node.
```

Comment:
This _will_ be true for the general case. However, right now, it would likely be confusing to read this sentence but see no way to give that desired type.

Perhaps we are simply clear about this, right now?

```suggestion
Explicitly configures the type of the annotated node. Currently, the only supported configuration is whether to allow any type or not.
```